Shared/UI/Views/DisplayableImageView.swift

```diff
@@ -149,14 +152,11 @@ class DisplayableImageView: View {
                 }
                 self.animationView.currentProgress = AnimationProgressTime(progress)
             }
-        } completed: { [weak self] image, error, cacheType, url in
-            guard let `self` = self, let downloadedImage = image else {
-                self?.showResult(for: nil)
-                return
-            }
+        })
 
-            self.showResult(for: downloadedImage)
-        }
+        guard !Task.isCancelled else { return }
```

Comment:
If the task was cancelled in the middle of the download, then don't update the image.